remove the generic min/maxTechLevelToBuild restrictions

llunak opened the pull request:
This breaks e.g. harp. It for some reason has minTechLevelToBuild set (a bug?), but it's actually made at a forge, so it perfectly possible for tribals to research and "build" it.

Since all vanilla items with minTechLevelToBuild have also a research requirement (with the exception with a horseshoes pin, which is really just a stick), it should be sufficient to rely just on research. Better err on the safe side.

emipa606 commented:
Perhaps it should check if the item has a research-requirement and only do the check if it hasnt?

llunak commented:
What would be the logic of that? I understand the mod as "cannot repair things until researched", but that would not apply here, as it would not be possible to repair them ever. And that's assuming it's even possible to get such items, which AFAICT is not the case at least with vanilla.

emipa606 commented:
There are ways of increasing the tech-level of the player faction. That was in fact one of the reasons for making the mod, to block the continued use of things until the faction has an appropriate understanding of the tech required. 
But I can probably add it as an option if it does not fit your playstyle. I usually do that anyway with major changes like this.

llunak committed:
remove the generic min/maxTechLevelToBuild restrictions if research-unlocked

Otherwise e.g. a harp is broken. It for some reason has minTechLevelToBuild
set (a bug?), but it's actually made at a forge, so it perfectly possible
for tribals to research and "build" it.

llunak commented:
Fair enough.

llunak commented:
Ping?

emipa606 commented:
I have it on the todolist
